### PR TITLE
[SYCL] [L0] Remove signaled events from wait list by default

### PR DESCRIPTION
**Pull request by asudarsa**
Signed-off-by: Arvind Sudarsanam <arvind.sudarsanam@intel.com>

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -756,7 +756,8 @@ pi_result _pi_queue::executeOpenCommandList() {
 }
 
 static const bool FilterEventWaitList = [] {
-  const bool RetVal = std::getenv("SYCL_PI_LEVEL_ZERO_FILTER_EVENT_WAIT_LIST");
+  const char *Ret = std::getenv("SYCL_PI_LEVEL_ZERO_FILTER_EVENT_WAIT_LIST");
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:759]: Please document it in https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md

**Comment by smaslov-intel** [comment]
Let's run as much as possible pre-commit testing for this change.

File: sycl/plugins/level_zero/pi_level_zero.cpp (hunk 1 above)
Comment (asudarsa) [reply sycl/plugins/level_zero/pi_level_zero.cpp:759]: Done. Thanks.

**Review by smaslov-intel — APPROVED** [review]
LGTM, please make sure no regression from the summary pre-commit testing before merging.

**Comment by bader** [comment]
> LGTM, please make sure no regression from the summary pre-commit testing before merging.

The tests pass for original commit and follow-up commits modified documentation only, so I think it's okay.

**Pull request merged**